src/Factory/Request/OrderAddressFactory.php

```diff
@@ -52,4 +52,19 @@ private function getCountryName(AddressInterface $address, ?string $localeCode):
 
         return null;
     }
+
+    private function getFullStreetName(AddressInterface $address): ?string
+    {
+        $fullStreetName = $address->getStreet();
+
+        if (method_exists($address, 'getHouseNumber') && $address->getHouseNumber() !== null) {
+            $fullStreetName .= ' ' . $address->getHouseNumber();
+        }
+
+        if (method_exists($address, 'getApartmentNumber') && $address->getApartmentNumber() !== null) {
+            $fullStreetName .= ' ' . $address->getApartmentNumber();
```

Comment:
If HouseNumber and ApartmentNumber exists, the separator should be a `-`. For i.e. `Vilniaus g. 12-55`